src/odemis/dataio/csv.py

```diff
@@ -45,159 +45,260 @@ def export(filename, data):
+        unit = "ps"
+        time_tuples = [(s, d) for s, d in zip(time_range, data)]
+        headers = ['# Time (ps)', 'intensity']
+    else:
+        logging.info("Exporting chronogram without time list information")
+        time_tuples = data.reshape(data.shape[0], 1)
+        headers = ['# intensity']
+
+    with open(filename, 'w', newline='') as fd:
+        csv_writer = csv.writer(fd)
+        csv_writer.writerow(headers)
+        csv_writer.writerows(time_tuples)
+
+    fd.close()
+
+
+def _export_angle_data(data, filename):
+    """
+    This will export the angle data to a CSV file using the acquired data,
+    and filename title in its creation.
+    :param data:
+    :param filename:
+    :return:
+    """
+    angle_range, unit_a = spectrum.get_angle_range(data)
+    if unit_a == "rad":
+        unit_a = "°"
+        angle_range = [math.degrees(theta) for theta in angle_range]  # Convert radians to degrees
+        angle_tuples = [(s, d) for s, d in zip(angle_range, data)]
+        headers = ['# Angle (°)', 'intensity']
+    else:
+        logging.info("Exporting angle without theta list information")
+        angle_tuples = data.reshape(data.shape[0], 1)
+        headers = ['# intensity']
+
+    with open(filename, 'w', newline='') as fd:
+        csv_writer = csv.writer(fd)
+        csv_writer.writerow(headers)
+        csv_writer.writerows(angle_tuples)
+
+    fd.close()
+
+
+def _export_ar_data(data, filename):
+    """
+    This will export the AR data to a CSV file using the acquired data,
+    and filename title in its creation.
+    :param data:
+    :param filename:
+    :return:
+    """
+    # Data should be in the form of (Y+1, X+1), with the first row and column the angles
+    with open(filename, 'w', newline='') as fd:
+        csv_writer = csv.writer(fd)
+
+        # TODO: put theta/phi angles in metadata? Read back from MD theta/phi and then add as additional line/column
+        # add the phi and theta values as an extra line/column in order to be displayed in the csv-file
+        # attach theta as first column
+        theta_lin = numpy.linspace(0, math.pi / 2, data.shape[0])
+        data = numpy.append(theta_lin.reshape(theta_lin.shape[0], 1), data, axis=1)
+        # attach phi as first row
+        phi_lin = numpy.linspace(0, 2 * math.pi, data.shape[1] - 1)
+        phi_lin = numpy.append([[0]], phi_lin.reshape(1, phi_lin.shape[0]), axis=1)
+        data = numpy.append(phi_lin, data, axis=0)
+
+        # Set the 'header' in the 0,0 element
+        first_row = ['theta\\phi[rad]'] + [d for d in data[0, 1:]]
+        csv_writer.writerow(first_row)
+        # dump the array
+        csv_writer.writerows(data[1:, :])
+
+    fd.close()
+
+
+def _export_spectrum_line_data(data, filename):
+    """
+    This will export the line data to a CSV file using the acquired data,
+    and filename title in its creation.
+    :param data:
+    :param filename:
+    :return:
```

Comment:
I'm not super convinced of having the parameters defined in the docstring, without actually any definition. I think you could drop them. At least, I'd suggest to drop the `return` as the function doesn't return anything.

Reply:
Considered the comments of Thera I choose to keep all of the docstrings with added definitions now. Indeed all return parameters have been deleted as well as the acq_type parameters.